tests/components/tessie/test_config_flow.py

```diff
@@ -0,0 +1,310 @@
+"""Test the Tessie config flow."""
+
+from unittest.mock import patch
+
+from homeassistant import config_entries
+from homeassistant.components.tessie.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from .common import (
+    ERROR_AUTH,
+    ERROR_CONNECTION,
+    ERROR_UNKNOWN,
+    TEST_DATA,
+    TEST_VEHICLES,
+    URL_VEHICLES,
+    setup_platform,
+)
+
+from tests.common import MockConfigEntry
+from tests.test_util.aiohttp import AiohttpClientMocker
+
+
+async def test_form(hass: HomeAssistant, aioclient_mock: AiohttpClientMocker) -> None:
+    """Test we get the form."""
+
+    aioclient_mock.get(
+        URL_VEHICLES,
+        text=TEST_VEHICLES,
+    )
```

Comment:
We should be mocking the library in tests, and the protocol level testing happens in the library itself.  We don't want the HA test to have to know about the protocol details or how it talks to the api, only how HA talks to the library

Reply:
So I tried for hours to mock this API, and couldn't for the life of me get it to work.

`with patch(
        "tessie_api.get_state_of_all_vehicles",
        return_value=TEST_VEHICLES,
    ):`

I looked at how other integrations do it, but I am struggling to understand how pytest fixtures and magicmock work.

If you can point me to an example of what you expect should work, or even just give me 1 working patch/mock ill reproduce that across the tests.

Reply:
something like this should work
```python
from datetime import timedelta
from tests.common import async_fire_time_changed
from homeassistant.util import dt as dt_util

with patch("homeassistant.components.tessie.coordinator.get_state_of_all_vehicles", return_value=DATA):
    async_fire_time_changed(hass, dt_util.utcnow() + timedelta(seconds=15))
    await hass.async_block_till_done()
```

Reply:
Ok, thats working! Rewriting all the tests now.